src/main/java/io/jenkins/plugins/jfrog/JfStep.java

```diff
@@ -294,4 +321,43 @@ public boolean isApplicable(Class<? extends AbstractProject> jobType) {
             return true;
         }
     }
+
+    String getJfrogCliVersion(Launcher.ProcStarter launcher) throws IOException, InterruptedException {
+        if (this.currentCliVersion != null && !this.currentCliVersion.isEmpty()) {
+            return this.currentCliVersion;
+        }
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        ArgumentListBuilder builder = new ArgumentListBuilder();
+        builder.add(jfrogBinaryPath).add("-v");
+        int exitCode = launcher
+                .cmds(builder)
+                .pwd(launcher.pwd())
+                .stdout(outputStream)
+                .join();
+        if (exitCode != 0) {
+            throw new IOException("Failed to get JFrog CLI version: " + outputStream.toString(StandardCharsets.UTF_8));
+        }
+        String versionOutput = outputStream.toString(StandardCharsets.UTF_8).trim();
+        String[] versionParts = versionOutput.split(" ");
+        if (versionOutput.isEmpty() || (versionParts.length < 3)) {
+            throw new IOException(String.format("Failed to parse JFrog CLI version. CLI Output: %s", versionOutput));
+        }
+        // Based on the output, the version should be located at the second index
+        return versionParts[2];
+    }
+
+    boolean isCliVersionGreaterThanOrEqual(String currentVersion, String targetVersion) {
+        String[] currentParts = currentVersion.split("\\.");
+        String[] targetParts = targetVersion.split("\\.");
+        for (int i = 0; i < Math.min(currentParts.length, targetParts.length); i++) {
+            int currentPart = Integer.parseInt(currentParts[i]);
+            int targetPart = Integer.parseInt(targetParts[i]);
+            if (currentPart > targetPart) {
+                return true;
+            } else if (currentPart < targetPart) {
+                return false;
+            }
+        }
+        return currentParts.length >= targetParts.length;
+    }
```

Comment:
Let's use Version from org.jfrog.build.client. This way we can store currentCliVersion and MIN_CLI_VERSION_PASSWORD_STDIN as "Version" instead strings, and use isAtLeast for comparison.
```suggestion
```

For reference see: https://github.com/jfrog/jenkins-jfrog-plugin/blob/4e451f6a8bd91b4af27bdae7e8d4f77f371393a7/src/main/java/io/jenkins/plugins/jfrog/ArtifactoryInstaller.java#L95